src/config/cli.rs

```diff
@@ -87,11 +87,15 @@ pub struct Cli {
     /// Show gas details information.
     pub show_gas_details: Option<ShowGasDetails>,
 
-    #[arg(long, help_heading = "Debugging Options")]
+    #[arg(long, default_missing_value = "true", num_args(0..=1), help_heading = "Debugging Options")]
     /// If true, the tool will try to resolve ABI and topic names for better readability.
     /// May decrease performance.
     pub resolve_hashes: Option<bool>,
 
+    #[arg(long, default_missing_value = "true", num_args(0..=1), help_heading = "Debugging Options")]
+    /// If true, the tool will only show contract logs.
+    pub show_only_contract_logs: Option<bool>,
```

Comment:
I don't particularly understand the purpose of `show_only_contract_logs`. Why does hardhat need all other logs to be silenced? 

Ideally we shouldn't use `show_only_contract_logs` to conditionally enable / disable all other logs. This is what `quiet` should do. If Hardhat absolutely needs ALL other logs to be silenced then it should just use `--quiet`. We can then by default have console logs ignore this flag, and be used alongside `--disable-console-log` if devs want complete silence. As they also have:

```
--disable-console-log
   Disable printing of `console.log` invocations to stdout
    
   [aliases: no-console-log]
```